platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/DependencyGraphBuilder.java

```diff
@@ -187,6 +187,11 @@ private void traverseGraph(final ResolveState resolveState) {
                     continue;
                 }
 
+                if (node.getComponent().isRejected()) {
```

Comment:
Without this change, there is still an infinite loop because we keep processing rejected but "selected" nodes.

Reply:
Change is now removed. Instead, since we clear conflicts when handling them we should not create or process a conflict where all remaining participants are `rejected`. Because it means we handled that conflict before and there was no solution.